analyses/cms-open-data-ttbar/utils/__init__.py

```diff
@@ -153,3 +194,386 @@ def get_data_rootfiles_uri(self, query, as_signed_url=True, title="Untitled"):
+
+        ax1.set_ylim([0, y_max * 1.5])  # 50% headroom
+        ax3.set_ylim([0, y_max * 1.5])  # 50% headroom
+    
+        # figure label (region name)
+        at = mpl.offsetbox.AnchoredText(label_prefit, loc="upper left", frameon=False, prop={"fontsize": "large", "linespacing": 1.5})
+        ax1.add_artist(at)
+        at = mpl.offsetbox.AnchoredText(label_postfit, loc="upper left", frameon=False, prop={"fontsize": "large", "linespacing": 1.5})
+        ax3.add_artist(at)
+        
+        # MC contributions in inverse order, such that first legend entry corresponds to
+        # the last (highest) contribution to the stack
+        all_containers = mc_containers_prefit[::-1] + [mc_unc_container_prefit, data_container_prefit]
+        all_labels = mc_labels[::-1] + ["Uncertainty", data_label]
+        ax1.legend(all_containers, all_labels, frameon=False, fontsize="large", loc="upper right")
+        all_containers = mc_containers_postfit[::-1] + [mc_unc_container_postfit, data_container_postfit]
+        ax3.legend(all_containers, all_labels, frameon=False, fontsize="large", loc="upper right")
+        
+        ax1.set_xlim(bin_edges[0], bin_edges[-1])
+        ax1.set_ylabel("events")
+        ax1.set_xticklabels([])
+        ax1.set_xticklabels([], minor=True)
+        ax1.tick_params(axis="both", which="major", pad=8)  # tick label - axis padding
+        ax1.tick_params(direction="in", top=True, right=True, which="both")
+
+        ax3.set_xlim(bin_edges[0], bin_edges[-1])
+        ax3.set_xticklabels([])
+        ax3.set_xticklabels([], minor=True)
+        ax3.set_yticklabels([])
+        ax3.set_yticklabels([], minor=True)
+        ax3.tick_params(axis="both", which="major", pad=8)  # tick label - axis padding
+        ax3.tick_params(direction="in", top=True, right=True, which="both")
+
+        ax2.set_xlim(bin_edges[0], bin_edges[-1])
+        ax2.set_ylim([0.5, 1.5])
+        ax2.set_xlabel(histogram_dict_list_prefit[0]["variable"])
+        ax2.set_ylabel("data / model")
+        ax2.set_yticks([0.5, 0.75, 1.0, 1.25, 1.5])
+        ax2.set_yticklabels([0.5, 0.75, 1.0, 1.25, ""])
+        ax2.tick_params(axis="both", which="major", pad=8)
+        ax2.tick_params(direction="in", top=True, right=True, which="both")
+
+        ax4.set_xlim(bin_edges[0], bin_edges[-1])
+        ax4.set_ylim([0.5, 1.5])
+        ax4.set_xlabel(histogram_dict_list_postfit[0]["variable"])
+        ax4.set_yticklabels([])
+        ax4.set_yticklabels([], minor=True)
+        ax4.tick_params(axis="both", which="major", pad=8)
+        ax4.tick_params(direction="in", top=True, right=True, which="both")
+        
+        fig.show()
+
+        figs.append({"figure": fig, "region": channel_name})
+    
+    
+    return figs
+    
+# modified from generate_config in https://github.com/triton-inference-server/fil_backend/blob/main/qa/L0_e2e/generate_example_model.py
+def generate_triton_config(model_name, 
```

Comment:
Is this used anywhere?

Reply:
Missed it, this is used in the training notebook (I was wondering whether it is used for inference, but it is written out and then later used).